### ISSUE TITLE
Fetch Metadata from crossref

### ISSUE THREAD

**Issue by jayghoshter** [burtonator/polar-bookshelf#1030]
My use case for polar is as a library manager and annotator for the research articles I hoard over time. In other words, it's a replacement for Mendeley/Zotero. One major thing missing, then, is the fetching of metadata for journal articles.

I've been playing with this concept for a while now on my own and have thrown together a script that takes care of this for me: 

1. Extract DOI from pdf (embedded text or pdf-metadata)
2. If found, search on crossref
3. else: query the title of the document on crossref 
4. Parse JSON output from crossref and extract relevant information
5. Cache information and modify file / pdf-metadata. 

I wrote it in bash for my own use and it depends on tools like `jq` for JSON parsing and `fzf` to make life easier, exiftool to edit file metadata and so on. It's not much, but I'll post the script here if it's required. 

Thanks for the great effort!

**Comment by burtonator** [comment]
Yeah.. might as well.  We have to update our internal metadata model to have the proper fields.

One thing I'm considering doing is having our own model but then 'grafting' on external metadata as json or a string.  This way we can support something like bibtex or json from some 3rd party APIs and preserve the original.  

I talked to the Internet Archive and they're working on a better API (I think) that they call FatCat that pulls in all these APIs into a meta API.  So they have pubmed, crossref, unpaywall, etc.

We might use them instead as they would have far more coverage.  

Not sure how fast or slow this would be.

**Comment by jayghoshter** [comment]
> One thing I'm considering doing is having our own model but then 'grafting' on external metadata as json or a string. This way we can support something like bibtex or json from some 3rd party APIs and preserve the original.

Yeah this seems like a decent hack, but having at least the basic fields of Title, Author, and Date are important I think. And then there's Publisher, Journal and so on. 

> I talked to the Internet Archive and they're working on a better API (I think) that they call FatCat that pulls in all these APIs into a meta API. So they have pubmed, crossref, unpaywall, etc.

This sounds truly great.

Here's that script of mine: 
[ptd.txt](https://github.com/burtonator/polar-bookshelf/files/3258973/ptd.txt)
